src/pages/achievement/control/subcomponents/controlPanelTools/controlPanelUtils/PrerequisiteAdder.tsx

```diff
@@ -0,0 +1,68 @@
+import { Button } from '@blueprintjs/core';
+import React, { useState } from 'react';
+
+import { AchievementItem } from '../../../../../../features/achievement/AchievementTypes';
+import AchievementInferencer from '../../../..//dashboard/subcomponents/utils/AchievementInferencer';
+import AchievementSelector from './AchievementSelector';
+
+type PrerequisiteAdderProps = {
+  editableAchievement: AchievementItem;
+  setEditableAchievement: any;
+  inferencer: AchievementInferencer;
+  saveChanges: any;
+};
+
+function PrerequisiteAdder(props: PrerequisiteAdderProps) {
+  const { editableAchievement, setEditableAchievement, inferencer, saveChanges } = props;
+
+  const [isDialogOpen, setDialogOpen] = useState<boolean>(false);
+  const toggleDialogOpen = () => setDialogOpen(!isDialogOpen);
+
+  const nonPrerequisites = inferencer.listAvailablePrerequisites(editableAchievement.id);
+
+  const addPrerequisite = (prerequisiteID: number) => {
+    const newAchievement = editableAchievement;
+    newAchievement.prerequisiteIds.push(prerequisiteID);
+    return newAchievement;
+  };
+
+  /**
+   * If there are no more prerequisites to add to this achievement,
+   * it will be set to a default value of 0.
+   *
+   * Else, it will always take the first available prerequisite id.
+   */
+  const [addedPrerequisiteID, setAddedPrerequisiteID] = useState<number>(
+    nonPrerequisites.length === 0 ? 0 : nonPrerequisites[0]
```

Comment:
can 0 be a prerequisite Id? If so, then default value can't be 0

Reply:
Yea :x 

Then will change it to -1 ba

Reply:
If -1 doesnt look too clean, you can define a const noRequisiteId = -1 somewhere :)

Reply:
aights!

Reply:
Fixed!